### DISCUSSION TITLE
Creation of Circuit Termination in a script

### DISCUSSION THREAD

**Discussion by aascedu** [netbox-community/netbox#18269] [Help Wanted!]
Hello, I am trying to import circuit termination and I have this error because of duplicate elements :
```python
An exception occurred: IntegrityError: duplicate key value violates unique constraint "circuits_circuittermination_unique_circuit_term_side" DETAIL: Key (circuit_id, term_side)=(734, A) already exists.

Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.12/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
psycopg.errors.UniqueViolation: duplicate key value violates unique constraint "circuits_circuittermination_unique_circuit_term_side"
DETAIL:  Key (circuit_id, term_side)=(734, A) already exists.
```

However I don't see a "circuit_id" in the source code of netbox models for CircuitTermination.

Also, it is very weird that I get this type of error because I use .exists() before creating it like so :
```python
if not CircuitTermination.objects.filter(circuit=fibre, term_side="A", provider_network=provider_network).exists():
		CircuitTermination.objects.create(
			circuit=fibre,
			term_side="A",
			provider_network=provider_network
		)
```
I also used .count(), and it was the same results...

The behavior is not usual so I would be tempted to say this is an issue/bug but my impostor syndrome takes over so I'd rather say it is a poor programming error originating from my side. Thanks for any help ! Merry Xmas

**Comment by candlerb** [comment]
> However I don't see a "circuit_id" in the source code of netbox models for CircuitTermination.

https://github.com/netbox-community/netbox/blob/v4.1.10/netbox/circuits/models/circuits.py#L232-L236

It's implied in the Django ORM that if you have a foreign key "circuit", then the auto-generated column name is "circuit_id"

> Also, it is very weird that I get this type of error because I use .exists() before creating it like so :

The unique index is only on `(circuit_id, term_side)`, and not on `provider_network`, so try this instead:

```
if not CircuitTermination.objects.filter(circuit=fibre, term_side="A").exists():
```
